src/orchestrator-kubernetes/src/cloud_resource_controller.rs

```diff
@@ -103,6 +106,63 @@ impl CloudResourceController for KubernetesOrchestrator {
         Ok(endpoints)
     }
 
+    async fn watch_vpc_endpoints(&self) -> BoxStream<'static, VpcEndpointEvent> {
+        let stream = watcher(self.vpc_endpoint_api.clone(), watcher::Config::default())
+            .touched_objects()
+            .filter_map(|object| async move {
+                match object {
+                    Ok(vpce) => {
+                        if let Some(connection_id) =
+                            mz_cloud_resources::id_from_vpc_endpoint_name(&vpce.name_any())
+                        {
```

Comment:
These nested if-else's can be avoided by doing things like:

```
let connection_id = mz_cloud_resources::id_from_vpc_endpoint_name(&vpce.name_any())?
```

question mark should work with options

results can be turned into options like:

```
object.map_err(|e| tracing::warn!("{}", e)).ok()?
```